common/libraries/sql-lib/src/main/java/org/eclipse/dataspaceconnector/sql/ResultSetMapper.java

```diff
@@ -0,0 +1,36 @@
+/*
+ *  Copyright (c) 2021 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.sql;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * Component capable of mapping {@link ResultSet} to e.g. POJO.
+ *
+ * @param <T> generic type param of the resulting type
+ */
+@FunctionalInterface
+public interface ResultSetMapper<T> {
+
+    /**
+     * Maps a sql result set into a object.
+     *
+     * @param resultSet containing columns of a row
+     * @return result
+     * @throws SQLException if something went wrong
+     */
+    T mapResultSet(ResultSet resultSet) throws SQLException;
```

Comment:
Must disagree, this is utility functionality and not part of any service extension. The integrating service extension has to handle those exception.

Reply:
The `ResultSetMapper` is only used in the `SqlQueryExecutor`, which will in turn likely be used in `*store` implementations, where try-catching the same SQL exception is nothing but a nuisance, because they will be rethrown as `EdcExceptions` anyway, so we'd only introduce multiple identical try-catch blocks.

On a more general note: checked exceptions should be wrapped in an unchecked exception as soon as possible. I know checked exceptions are an ambivalent topic, but a design decision has been made to not use them.

Reply:
@paullatzelsperger Alright, I'll give it a try, I'll introduce that `EdcPersistenceException` then. Where shall it come from? Shall I create a new class within `:extensions:sql:common` or directly in the `:spi` module?

Reply:
I think it can be in the `:spi` module, most likely `:spi:core-spi`. Ideally, all the other persistence extensions will use it as well then.

Reply:
If we want other persistence extensions (eg cosmos) to also throw it, spi is the common denominator, right?
If we don't, then it doesn't need to go into spi.